File: backend/src/models/message.model.js
const mongoose = require('mongoose');

// In-trip chat between a customer and driver, scoped to one order. Real-time
// delivery happens over the same Socket.IO booking room already used for
// live GPS/status broadcasts (see sockets/tracking.socket.js); this is the
// durable copy backing history for whoever opens the chat screen later or
// was offline when a message arrived.
const messageSchema = new mongoose.Schema(
  {
    bookingId: { type: mongoose.Schema.Types.ObjectId, ref: 'Order', required: true, index: true },
    senderId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    senderRole: { type: String, enum: ['customer', 'driver'], required: true },
    text: { type: String, required: true },
  },
  { timestamps: true }
);

module.exports = mongoose.model('Message', messageSchema);

File: mobile-driver-app/lib/models/message_model.dart
class ChatMessage {
  final String id;
  final String bookingId;
  final String senderId;
  final String senderRole; // 'customer' | 'driver'
  final String text;
  final DateTime createdAt;

  ChatMessage({
    required this.id,
    required this.bookingId,
    required this.senderId,
    required this.senderRole,
    required this.text,
    required this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['_id'] as String? ?? json['id'] as String,
      bookingId: json['bookingId'] as String,
      senderId: json['senderId'] as String,
      senderRole: json['senderRole'] as String,
      text: json['text'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}
